### ISSUE TITLE
Parsing wrong values in javascript compared to C++ array values(pointers) in wasm javascript( emscripten interface)

### ISSUE THREAD

**Issue by vasanthhr** [emscripten-core/emscripten#9719]
Hello, I am getting different values when i print using CPP code and access the same values returned to javascript(emscriptten interface). please see the below code. 

defined variable  in cpp code as follows:

```
this->marks = std::vector<uint8_t>(136);  //size of 136


this below function returns a pointer.
//function1
int processImage(int landmark_number){
				// We return a 1D array on the heap with its length stashed as the first element
 for(int i = 0, j = 0; i < landmark_number; ++i, j += 2)
 {

                            marks_cur = getLm();
				float x = marks_cur(i, 0);
				float y = marks_cur(i, 1);
				this->marks[j] = x;
				this->marks[j+1] = y;

                                 std::cout<< x  << y   //console : this is cpp values
			}
			return (int)(this->marks.data());

 }

//function2
int getSize(){
   return marks.size();  //returns 136
}

```
I am trying to access in javascript as:

```
let ptr = processImage(68);
let len =getSize();
Lmarks = new Uint8Array(Module_BUILT.HEAPU8.buffer, ptr, len);
console.log(Lmarks ) // //console : this is returned cpp values and accessed in javascript values (but values are not matching with CPP std::cout values).


```

**Comment by VirtualTim** [comment]
So I found it a little hard to follow your code but here's the bit that look odd to me:
`return (int)(this->marks.data());` - you're casting a pointer (uint8_t*) to an int?

I feel like `processImage` should be returning a `uint8_t*`.

You could also do `Lmarks = Module_BUILT.HEAPU8.slice(ptr, ptr+len);` (or subarry instead of slice). Not saying this is "better", just wanted to make you aware.

**Comment by vasanthhr** [comment]
@VirtualTim , Thanks for your response.

I am able to solve this issue. And thanks for your response.

**Comment by VirtualTim** [comment]
Glad to hear it. If it's fixed can you close the issue?

**Comment by vasanthhr** [comment]
Yes, I closed it.
